core/domain/html_validation_service_test.py

```diff
@@ -1672,6 +1672,123 @@ def test_validate_math_tags_in_html_with_attribute_math_content(self):
+        extracts the required information from all the math tags.
+        """
+        html_string1 = (
+            '<p>Feedback</p><oppia-noninteractive-math math_content-with-v'
+            'alue="{&amp;quot;raw_latex&amp;quot;: &amp;quot;+,-,-,+'
+            '&amp;quot;, &amp;quot;svg_filename&amp;quot;: &amp;quot;&amp'
+            ';quot;}"></oppia-noninteractive-math>'
+            '<oppia-noninteractive-math math_content-with-value="{&amp;'
+            'quot;raw_latex&amp;quot;: &amp;quot;+,+,+,+&amp;quot;, &amp;'
+            'quot;svg_filename&amp;quot;: &amp;quot;&amp;quot;}"></oppia'
+            '-noninteractive-math>'
+            '<oppia-noninteractive-math math_content-with-value="{&amp;q'
+            'uot;raw_latex&amp;quot;: &amp;quot;(x - a_1)(x - a_2)(x - a'
+            '_3)...(x - a_n)&amp;quot;, &amp;quot;svg_filename&amp;quot;'
+            ': &amp;quot;&amp;quot;}"></oppia-noninteractive-math>')
+
+        expected_list_of_latex = [
+            '+,-,-,+', '+,+,+,+', '(x - a_1)(x - a_2)(x - a_3)...(x - a_n)']
+        expected_size = 45000
+        expected_largest_expression = '(x - a_1)(x - a_2)(x - a_3)...(x - a_n)'
+        size, largest_expression, list_of_latex = (
+            html_validation_service.
+            extract_math_rich_text_related_information_from_html(html_string1))
+        self.assertEqual(expected_size, size)
+        self.assertEqual(largest_expression, expected_largest_expression)
+        self.assertEqual(sorted(list_of_latex), sorted(expected_list_of_latex))
+
+        html_string2 = (
+            '<p>Feedback</p><oppia-noninteractive-math math_content-with-v'
+            'alue="{&amp;quot;raw_latex&amp;quot;: &amp;quot;\\\\frac{x}{y}'
+            '&amp;quot;, &amp;quot;svg_filename&amp;quot;: &amp;quot;&amp'
+            ';quot;}"></oppia-noninteractive-math>'
+            '<oppia-noninteractive-math math_content-with-value="{&amp;'
+            'quot;raw_latex&amp;quot;: &amp;quot;+,+,+,+(x^2)&amp;quot;, &amp;'
+            'quot;svg_filename&amp;quot;: &amp;quot;&amp;quot;}"></oppia'
+            '-noninteractive-math>'
+            '<oppia-noninteractive-math math_content-with-value="{&amp;q'
+            'uot;raw_latex&amp;quot;: &amp;quot;\\\\sqrt{x}&amp;quot;, &am'
+            'p;quot;svg_filename&amp;quot;: &amp;quot;&amp;quot;}"></opp'
+            'ia-noninteractive-math>')
+
+        expected_list_of_latex = [
+            '\\sqrt{x}', '\\frac{x}{y}', '+,+,+,+(x^2)']
+        expected_size = 23000
+        expected_largest_expression = '+,+,+,+(x^2)'
+        size, largest_expression, list_of_latex = (
+            html_validation_service.
+            extract_math_rich_text_related_information_from_html(html_string2))
+        self.assertEqual(expected_size, size)
+        self.assertEqual(largest_expression, expected_largest_expression)
+        self.assertEqual(sorted(list_of_latex), sorted(expected_list_of_latex))
+
+
+
+        html_string_with_no_math = (
```

Comment:
You can put this in a separate test function if you are trying to separate it from the other two.

Reply:
done